### PR TITLE
feat(avo): Add ActiveStorage resources (Blob, Attachment, VariantRecord)

### LINKED ISSUE #7132: Feature: Add file storage management resources to the Avo admin dashboard.
### Is your feature request related to a problem? Please describe

As part of the ongoing RailsAdmin → Avo admin migration  file storage management resources are not yet available in the Avo dashboard.

### Describe the solution you'd like

Addition of resources
ActiveStorage::Blob
Stores metadata about uploaded files such as filename, size, and content type.

ActiveStorage::Attachment c
Connects uploaded files to application records like users, projects, or other models.

ActiveStorage::VariantRecord
Stores processed file variants (such as resized images) generated by ActiveStorage.

### Describe alternatives you've considered

No alternative

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7132

 

#### Describe the changes you have made in this PR -
This PR adds Avo admin panel resources for Rails ActiveStorage file management, specifically:
- **ActiveStorage::Blob**: Manages uploaded file metadata (filename, content type, size, checksum)
- **ActiveStorage::Attachment**: Handles polymorphic file attachments to User and Project models
- **ActiveStorage::VariantRecord**: Tracks image variants and transformations

All resources follow the minimal approach - only resource files with field definitions, no filters or actions.
### Screenshots of the UI changes (If any) -
_Screenshots to be added after testing in admin panel_

---
## Code Understanding and AI Usage
**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

This PR exposes Rails ActiveStorage models in the Avo admin panel to provide visibility into file uploads and attachments.

**Implementation approach:**
- **Blob resource**: Displays file metadata fields (filename, content_type, byte_size, checksum, service_name, metadata) with read-only access. Includes associations to attachments and variant_records for tracking file usage.
- **Attachment resource**: Shows polymorphic relationships between blobs and their parent records (User or Project). Uses `record_type` and `record_id` fields to display which model owns each file.
- **VariantRecord resource**: Tracks image transformations/variants with blob association and variation_digest for identifying specific variants.
- **Blob controller**: Overrides default Avo behavior to block manual blob creation (shows flash message) since ActiveStorage automatically creates blobs when files are uploaded through the app.
---
## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

## Summary by CodeRabbit

* **New Features**
  * Added admin resources for attachments, blobs, and variant records, visible in the admin sidebar.
  * Lists are ordered for easier browsing (recent first).
  * Blob view shows image preview when applicable, and counts related attachments/variants.
  * Polymorphic and blob associations are searchable for quicker lookup.
  * Manual creation of blobs is blocked with an alert and redirect.

### CODE REVIEW

File: app/avo/resources/active_storage_variant_record.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  self.visible_on_sidebar = true
+  self.index_query = -> { query.order(id: :desc) }
+
+  def fields
+    field :id, as: :id, link_to_record: true
+    field :blob, as: :belongs_to, searchable: true
+    field :variation_digest, as: :text
+
+    # Image field for variant
+    field :image, as: :file, is_image: true
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/avo/resources/active_storage_variant_record.rb]: fixed and outdated @coderabbitai

**Comment by naman79820** [comment]
ready for review @tachyons

**Comment by tachyons** [comment]
This is not needed for now
